django/forms/models.py

```diff
@@ -1284,7 +1284,11 @@ def to_python(self, value):
                 value = getattr(value, key)
             value = self.queryset.get(**{key: value})
         except (ValueError, TypeError, self.queryset.model.DoesNotExist):
-            raise ValidationError(self.error_messages['invalid_choice'], code='invalid_choice')
+            raise ValidationError(
+                self.error_messages['invalid_choice'],
+                code='invalid_choice',
+                params={"value": value}
```

Comment:
I would only pass a `value`, without changing the default `invalid_choice` message, see a similar change for built-in validators 83fbaa92311dd96e330496a0e443ea71b9c183e2.

Reply:
You mean  something like this, 
```python
            raise ValidationError(
                _('Select a valid choice. "%(value)s" is not one of the available choices.'),
                code='invalid_choice',
                params={"value": value}
            )
```

Reply:
No, that would make it impossible to override the `invalid_choice` message. IMO we should revert changes to the `invalid_choice` message (L1191-L1192) and only pass a `value` to the `ValidationError`:
```python
raise ValidationError(
    self.error_messages['invalid_choice'],
    code='invalid_choice',
    params={"value": value},
)  
```

Reply:
Thats nice, but I would expect the `ModelChoiceField` behave like [`ModelMultipleChoiceField`](https://docs.djangoproject.com/en/3.1/ref/forms/fields/#modelmultiplechoicefield) which has **`invalid_choice`** with a [`value` attribute, by default](https://github.com/django/django/blob/83fbaa92311dd96e330496a0e443ea71b9c183e2/django/forms/models.py#L1301).

Reply:
In both situations, the developer will be able to customize the error message with ease and I think your suggestion requires only less work compared to my suggestion (code had broken many places because of that).